CONTRIBUTING.md

```
@@ -0,0 +1,59 @@
+pnpm lint
+pnpm test
+pnpm test:types
+```
+
+These run in CI on every PR and must pass before merge.
+
+## Commit Message Format
+
+This project follows the [Conventional Commits](https://www.conventionalcommits.org/) specification. All commit messages should be structured as follows:
+
+```
```

markdownlint-cli2 (0.23.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)